python/python/lance/fragment.py

```diff
@@ -90,3 +89,39 @@ def to_table(
         return self.scanner(
             columns=columns, filter=filter, limit=limit, offset=offset
         ).to_table()
+
+    def add_columns(
+        self,
+        value_func: Callable[[pa.RecordBatch], pa.RecordBatch],
```

Comment:
so the value_func's output will be the augmented Fragment that will then get written as the new version right? Going to assume the API will improve in subsequent PRs

Reply:
so the idea is that value_func will take of matching the input data and the merge data?

Reply:
the value_func will be "the producer" of the new data. This is similar to `pandas udf` in spark, so it takes in a "RecordBatch" as input, and create a same size "RecordBatch" as output